### DISCUSSION TITLE
FormState in resolver

### DISCUSSION THREAD

**Discussion by deskoh** [react-hook-form/react-hook-form#10814] [Q&A]
Is it possible to get the `FormState` in the resolver function?

Specifically, I would like to get the value of `submitCount` in the resolver function.

**Comment by Moshyfawn** [comment]
What's your use-case for reading `submitCount` in the resolver? The resolver isn't a good way to do it since it runs before submission, when `submitCount` has not been updated yet.

**Comment by deskoh** [comment]
The desired behavior is that during form entry, there will be business rules validating a set of inter-dependent text fields during `onChange`. On submit, there will be another set of validation rules (e.g. required fields etc.)

`resolver` is used for implementing the business rules validation with `mode: 'onChange'`, was thinking how to implement another set of 'pre-submit' validation rules.

Currently using the submit button `onClick` handler to add additional errors to the `error` object. But running into some weird behaviors. Will to replicate them in codesandbox soon.
